presto-main/src/main/java/com/facebook/presto/sql/planner/PlanFragment.java

```diff
@@ -144,6 +144,11 @@ public boolean isLeaf()
         return remoteSourceNodes;
     }
 
+    public boolean hasDefaultOutput()
```

Comment:
I do not like this method being here. What does it mean for fragment to have a default output? To me this is something specific for aggregation which should not leak here.